whatshap/cli/haplotagphase.py

```diff
@@ -113,18 +117,27 @@ def run_haplotagphase(
             sample_to_super_reads, sample_to_components = (dict(), dict())
             for sample in vcf_reader.samples:
                 logger.info(f"Processing sample {sample}")
+                genotypes = variant_table.genotypes_of(sample)
                 with timers("read-bam"):
-                    reads, _ = phased_input_reader.read(chromosome, variant_table.variants, sample)
+                    reads, _ = phased_input_reader.read(
+                        chromosome, variant_table.variants, sample, predicted_genotypes=genotypes
+                    )
                 phases = variant_table.phases_of(sample)
-                genotypes = variant_table.genotypes_of(sample)
+
                 homozygous = dict()
                 change = dict()
                 phased = dict()
+                # mapping of detected variants to 0/1 and reversed mappings.
+                allele_to_id = defaultdict(dict)
+                id_to_allele = defaultdict(dict)
```

Comment:
note to self: When we enable MAVs in `whatshap phase` as well, we should encapsulate this "mapping of variants to 0/1 and back" into a class.